api/jest.utils.ts

```diff
@@ -51,6 +52,10 @@ export function superRequest(
     void req.set('Cookie', setCookies);
   }
 
+  if (headers) {
+    void req.set('Referer', headers.referer);
+  }
+
```

Comment:
Intellisense is rather good for stuff like this:

![image](https://github.com/freeCodeCamp/freeCodeCamp/assets/15801806/4b7582b7-51ab-460d-9cb4-b8cdb0139f3d)

In principle we could come up with a better type, but since we only need the one header I think this is enough.